include/beast/websocket/detail/invokable.hpp

```diff
@@ -125,15 +125,17 @@ class invokable
     void
     emplace(F&& f);
 
-    void
+    bool
     maybe_invoke()
     {
         if(base_)
         {
             auto const basep = base_;
```

Comment:
Nope, because a call to `(*base_)()` could cause `emplace()` to get called which would break invariants. `base_` has to be set to `nullptr` before the invocation takes place.